pkg/apis/core/validation/validation.go

```diff
@@ -5498,6 +5578,11 @@ func ValidatePodResize(newPod, oldPod *core.Pod, opts PodValidationOptions) fiel
 	allErrs := ValidateImmutableField(&newPod.ObjectMeta, &oldPod.ObjectMeta, fldPath)
 	allErrs = append(allErrs, validatePodMetadataAndSpec(newPod, opts)...)
 
+	// pods with pod-level resources cannot be resized
+	if isPodLevelResourcesSet(oldPod) || isPodLevelResourcesSet(newPod) {
```

Comment:
yes, since this is updating the created pod. If the pod was created with resources when feature gate was enabled, and then updated when feature gate is disabled, this should still fail.

Reply:
I think as long as the feature gate is disabled, the validation should ignore the field and pass same as previous behavior. Update should not fail if the field is set but feature gate is disabled.
/cc @thockin for verification.

Reply:
I should clarified this in my previous comment. We don't want to support in place pod resize with pod-level resources. Hence failing the validation here.